tests/unit/reconcile/connectors/test_snowflake.py

```diff
@@ -0,0 +1,249 @@
+from unittest.mock import MagicMock
+
+import pytest
+from pyspark.errors import PySparkException
+
+from databricks.labs.remorph.reconcile.connectors.snowflake import SnowflakeDataSource
+from databricks.labs.remorph.reconcile.recon_config import JdbcReaderOptions, Tables
+
+
+@pytest.fixture
+def snowflake_data_source():
+    """
+    This fixture creates a mock instance of the SnowflakeDataSource class for testing purposes.
+
+    The SnowflakeDataSource instance is created with a mock SparkSession and predefined source, workspace, and scope.
+    The _get_secrets method of the instance is also mocked to return predefined secrets based on the secret name.
+
+    Returns:
+        SnowflakeDataSource: A mock instance of the SnowflakeDataSource class.
+    """
+
+    # Create a mock SparkSession
+    pyspark_sql_session = MagicMock()
+    spark = pyspark_sql_session.SparkSession.builder.getOrCreate()
+
+    # Define the source, workspace, and scope
+    source = "snowflake"
+    ws = "ws"
+    scope = "scope"
+
+    # Create a mock instance of SnowflakeDataSource
+    SnowflakeDataSource_ = SnowflakeDataSource(source, spark, ws, scope)
+
+    # Mock the _get_secrets method to return predefined secrets
+    SnowflakeDataSource_._get_secrets = MagicMock()
```

Comment:
Don't mock/patch/override private fields. Inject them through constructor instead

Reply:
Got it! I have made changes to the code to modify the way secrets are accessed and also injected mocked data through the mocked constructor.